main.go

```diff
@@ -118,12 +127,40 @@ func argsParse(argv []string) (finder.Args, error) {
 		args.KubeletConfigFile = kubeletConfigPath
 	}
 	args.SysfsRoot = arguments["--sysfs"].(string)
-	runtime := arguments["--container-runtime"].(string)
-	if !(runtime == "containerd" || runtime == "cri-o") {
-		return args, fmt.Errorf("invalid --container-runtime specified")
+
+	if source, ok := arguments["--source"].(string); ok {
+		args.Source = source
 	}
-	args.ContainerRuntime = runtime
-	args.CRIEndpointPath = arguments["--cri-path"].(string)
+	if args.Source != PodResourceSource && args.Source != CRISource {
+		return args, fmt.Errorf("invalid --source specified")
+
+	} else if args.Source == PodResourceSource {
+		//podresource source
+		if path, ok := arguments["--podresources-socket"].(string); ok {
+			args.PodResourceSocketPath = path
+		}
+		if _, ok := arguments["--cri-socket"].(string); ok {
+			return args, fmt.Errorf("No need to specify CRI socket path in case pod-resource-api is specified as the source")
+		}
+		if _, ok := arguments["--container-runtime"].(string); ok {
+			return args, fmt.Errorf("No need to specify container runtime in case pod-resource-api is specified as the source")
+		}
+	} else {
+		//cri source
+		if path, ok := arguments["--cri-socket"].(string); ok {
+			args.CRISocketPath = path
+		}
+		if _, ok := arguments["--podresources-socket"].(string); ok {
+			return args, fmt.Errorf("No need to specify Pod Resource socket path in case CRI is specified as the source")
+		}
+		runtime := arguments["--container-runtime"].(string)
+		if !(runtime == ContainerdRuntime || runtime == CRIORuntime) {
+			return args, fmt.Errorf("invalid --container-runtime specified")
+		}
+		args.ContainerRuntime = runtime
```

Comment:
a little hard to follow but we can clean it up later

Reply:
Added a few comments to explain the code, hope it helps.